Fix removeAt

nielsenko opened the pull request:
Apparently we never wired up and tested `removeAt`.

This fixes #883

nirinchev commented:
Isn't `removeAt` also implemented in `ListMixin`? The issue is that their implementation wouldn't work with Realm (the same way `removeAt` doesn't work) because they're manipulating the length of the list.

desistefanova commented on test/list_test.dart:
```diff
@@ -361,4 +361,27 @@ Future<void> main([List<String>? args]) async {
+    expect(realm.write(() => team.players.removeAt(0)), alice);
+    expect(team.players.length, 2);
+    expect(team.players, [bob, dan]);
+  });
```
We need a check that removed items still exist in the ream, detached from the list. Similar to the tests for Clear https://github.com/realm/realm-dart/blob/77e730551b01a12c49c47367b16090a47942a557/test/list_test.dart#L98

nielsenko replied:
I have added checks for that as well

nielsenko commented:
> Isn't `removeAt` also implemented in `ListMixin`? The issue is that their implementation wouldn't work with Realm (the same way `removeAt` doesn't work) because they're manipulating the length of the list.

Yes. I have implemented `length` setter to truncate, if `newLength` is larger than `length`, and implemented `removeRange` to support it.

I have also added tests for:
> * `contains`
> * `set length` -> should throw
> * `removeWhere`
> * `retainWhere`
> * `removeLast`
> * `removeRange`
> * `replaceRange`
